src/pages/resultsView/survival/SurvivalUtil.tsx

```diff
@@ -85,8 +85,50 @@ export const survivalClinicalDataNullValueSet = new Set([
     'na',
 ]);
 
+// Concept of Number at risk:
+// In order to calculate survival probability using the Kaplan-Meier product limit method
+// We need to know how many individuals were still accounted for in the study that had not yet experienced the event of interest.
+// Therefore, the number at risk at any specific time point will be equal to the total number of subjects remaining in the study,
+// including any individuals that experience the event of interest or individuals that are censored at this time point.
+export function getNumPatientsAtRisk(
+    // In order to calculate correct result
+    // Patient survivals need to be sorted by months and status
+    // First sort by month in asc order (smaller number to the front)
+    // Then sort by status in desc order (which means if status are '1's, then go to the front, '0's are right after '1's in the same time stamp)
+    sortedPatientSurvivals: Pick<PatientSurvival, 'entryMonths' | 'months'>[]
+): number[] {
+    // returns an array in the same order as patientSurvivals
+    const entryTimes = _.sortBy(sortedPatientSurvivals.map(s => s.entryMonths));
+
+    // When we see one entry time is not zero, we consider the data is left truncated
+    if (_.some(entryTimes, entryTime => entryTime !== 0)) {
+        // Calculate number of patients at risk for data with entry time (left truncation)
+        return _.reduce(
```

Comment:
i think this is actually a mapping operation because you are always pushing to the aggregator for every iteration.  in that case, the output array will always have the same number of elements as the input array.  reduce should be used only if there is a conditional or if the aggregator has a different type than the input (e.g. array -> key/value or summation).

Reply:
apologies if i suggested before using reduce.  but the semantics of these methods really helps a reader understand what is going on.